src/snowflake/snowpark/mock/_connection.py

```diff
@@ -136,15 +136,32 @@ def write_table(
                 # Fix append by index
                 if name in self.table_registry:
                     target_table = self.table_registry[name]
+                    input_schema = table.columns.to_list()
+                    existing_schema = target_table.columns.to_list()
 
-                    if len(table.columns.to_list()) != len(
-                        target_table.columns.to_list()
+                    if len(input_schema) <= len(existing_schema) and (
+                        all(
+                            target_table[col].sf_type.nullable
+                            for col in (existing_schema[len(input_schema) :])
+                        )
                     ):
+                        for col in existing_schema[len(input_schema) :]:
+                            table[col] = None
+                            table.sf_types[col] = target_table[col].sf_type
+                    else:
                         raise SnowparkLocalTestingException(
                             f"Cannot append because incoming data has different schema {table.columns.to_list()} than existing table { target_table.columns.to_list()}"
                         )
 
-                    table.columns = target_table.columns
+                    # This should only happen if column_order == "name" but we have no way to check the column_order
```

Comment:
@sfc-gh-stan Still need your help checking the column_order parameter and accordingly performing these operations.

Reply:
Because of not having a `column_order == "name"` check test case: `tests/integ/test_dataframe.py::test_create_dataframe_string_length` is failing right now. That's the only test case that is failing.